src/WpBookingsUiModule.php

```diff
@@ -0,0 +1,499 @@
+
+        $this->_attach('admin_menu', function () use ($templateManager, $c) {
+            $this->_adminMenu($templateManager, $c);
+        });
+
+        $this->_attach('wp_ajax_set_' . $c->get('wp_bookings_ui/screen_options/key'), function () use ($c) {
+            $data = json_decode(file_get_contents('php://input'), true);
+            $statuses = $data['statuses'];
+            $this->_setScreenStatuses($c, $statuses);
+        });
+    }
+
+    /**
+     * Check current screen is screen where EDDBK UI should be rendered.
+     *
+     * @return bool
+     */
+    protected function _onAppPage()
+    {
+        return in_array(get_current_screen()->id, [
+            $this->bookingsPageId,
+            $this->metaboxPageId,
+        ]);
+    }
+
+    /**
+     * Check current screen is bookings page.
+     *
+     * @return bool
+     */
+    protected function _onBookingsPage()
+    {
+        return get_current_screen()->id === $this->bookingsPageId;
+    }
+
+    /**
+     * Get app state for booking page.
+     *
+     * @param ContainerInterface $c
+     * @return array
+     */
+    protected function _getBookingsAppState($c)
+    {
+        /* @var ControllerInterface $controller */
+        $controller = $c->get('eddbk_services_controller');
+        $services = iterator_to_array($controller->get());
+
+        return [
+            /*
+             * All available statuses in application.
+             */
+            'statuses' => $this->_trigger('eddbk_bookings_statuses', [
+                'statuses' => [
+                    "draft" => $this->__('Draft'),
+                    "in_cart" => $this->__('Cart'),
+                    "pending" => $this->__('Pending'),
+                    "approved" => $this->__('Approved'),
+                    "rejected" => $this->__('Rejected'),
+                    "scheduled" => $this->__('Scheduled'),
+                    "cancelled" => $this->__('Cancelled'),
+                    "completed" => $this->__('Completed')
+                ]
+            ])->getParam('statuses'),
+
+            /*
+             * Statuses that enabled for filtering bookings.
+             */
+            'screenStatuses' => $this->_trigger('eddbk_bookings_screen_statuses', [
+                'screenStatuses' => $this->_getScreenStatuses($c)
+            ])->getParam('screenStatuses'),
+
+            /*
+             * List of available services.
+             */
+            'services' => $this->_trigger('eddbk_bookings_services', [
+                'services' => $services
+            ])->getParam('services'),
+
+            'statusesEndpoint' => $c->get('wp_bookings_ui/screen_options/endpoint'),
+
+            'endpointsConfig' => $this->_prepareEndpoints($c->get('wp_bookings_ui/endpoints_config'))
+        ];
+    }
+
+    /**
+     * Prepare endpoints for consuming in the UI
+     *
+     * @param $endpointsConfig
+     * @return array
+     */
+    protected function _prepareEndpoints($endpointsConfig)

```

Comment:
This looks like quite generic code that can, based on a config structure convention, work for any module. @mecha, thoughts?

Reply:
Not sure what exactly this code is trying to achieve. Why does the UI module need to register WP REST API endpoints?

Reply:
@mecha, this method is not for registering endpoints. Using this method I'm preparing endpoints for using in JS. I cannot wrap it in the `rest_url(..)` inside `config.php` so I'm doing it in the module.